Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7247,6 +7265,2013 @@ def panorama_install_file_content_update_command(args: dict):
+                        r.name = template_name
+                        result.append(r)
+
+        return result
+
+    @staticmethod
+    def push_all(
+            topology: Topology,
+            device_filter_str: str = None,
+            device_group_filter: List[str] = None,
+            template_stack_filter: List[str] = None
+    ) -> List[PushStatus]:
+        """Pushes the pending configuration from Panorama to the firewalls. This is an async function,
+        and will only push if there is config pending."""
+        result = []
+
+        for device in topology.active_top_level_devices(device_filter_str):
+            # Get the relevent DGs and Templates to push.
+            device_groups = PanoramaCommand.get_device_groups(topology, resolve_host_id(device))
+            device_group_names = set([x.name for x in device_groups])
+            template_stacks = PanoramaCommand.get_template_stacks(topology, resolve_host_id(device))
+            template_stack_names = set([x.name for x in template_stacks])
+
+            if device_group_filter:
+                device_group_names = set([x for x in device_group_names if x in device_group_filter])
+
+            if template_stack_filter:
+                template_stack_names = set([x for x in template_stack_names if x in template_stack_filter])
+
+            for dg_name in device_group_names:
+                device_group_commit = PanoramaCommitAll(
+                    style="device group",
+                    name=dg_name
+                )
+                result_job_id = device.commit(cmd=device_group_commit)  # type: ignore
+                result.append(PushStatus(
+                    hostid=resolve_host_id(device),
+                    commit_type="devicegroup",
+                    name=dg_name,
+                    job_id=result_job_id,  # type: ignore
+                    commit_all_status="Initiated",
+                    device_status="",
+                    device=""
+                ))
+
+            for template_name in template_stack_names:
+                template_stack_commit = PanoramaCommitAll(
+                    style="template stack",
+                    name=template_name
+                )
+                result_job_id = device.commit(cmd=template_stack_commit)  # type: ignore
+                result.append(PushStatus(
+                    hostid=resolve_host_id(device),
+                    commit_type="template-stack",
+                    name=template_name,
+                    job_id=result_job_id,  # type: ignore
+                    commit_all_status="Initiated",
+                    device_status="",
+                    device=""
+                ))
+
+        return result
+
+    @staticmethod
+    def get_push_status(topology: Topology, match_job_ids: List[str] = None) -> List[PushStatus]:
+        """Retrieves the status of a Panorama Push, using the given job ids."""
+        result: List[PushStatus] = []
```

Comment:
why not call it push_statuses or something similar?

Reply:
See above comments about the use of result as a variable name